### ISSUE TITLE
Have a more precise error message for files outside a project

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [palantir/tslint#3377]
> ```typescript
> const sourceFile = this.program.getSourceFile(fileName);
> if (sourceFile === undefined) {
>     const INVALID_SOURCE_ERROR = dedent`
>         Invalid source file: ${fileName}. Ensure that the files supplied to lint have a .ts, .tsx, .d.ts, .js or .jsx extension.
>     `;
>     throw new FatalError(INVALID_SOURCE_ERROR);
> }
> ```

That error message implies the file is corrupt, particularly from having an improper extension. It may be just that the file isn't listed in the project.

**Comment by ajafff** [comment]
If we are talking about the error you see when running from CLI, #3313 will fix this.

The error you get when using the API could be amended. But why do you even get that error?

**Comment by JoshuaKGoldberg** [comment]
I made the user error of trying lint a file not listed in a `tsconfig.json`:

```shell
tslint --config tslint.json --project tsconfig.json src/**/*.ts scripts/not-in-tsconfig.js
```

**Comment by ajafff** [comment]
That's already tracked by #2208 and will be fixed by #3313
